src/Microsoft.DotNet.Interactive.SqlServer.Tests/MsSqlConnectionTests.cs

```diff
@@ -56,11 +56,18 @@ public async Task It_can_connect_and_query_data()
             events.Should().NotContainErrors();
 
             events.Should()
-                  .ContainSingle<DisplayedValueProduced>()
-                  .Which
-                  .FormattedValues
-                  .Should()
-                  .ContainSingle(f => f.MimeType == HtmlFormatter.MimeType);
+                .Contain(e => e is DisplayedValueProduced && e.As<DisplayedValueProduced>()
```

Comment:
We have an overload extension or two for `ContainSingle` that can be helpful here. Here's an example:

```cs
events
  .Should()  
  .NotContainErrors()
  .And.ContainSingle<DisplayedValueProduced>(e => e.FormattedValues.Any(f => f.MimeType == "text/html")) // <-- this is a filter   
  .Which.FormattedValues.Should().ContainSingle(f => f.MimeType == "text/html")
  .Which.Value.Should().Contain("...")
```

Reply:
(That might be able to be more concise but just wanted to call attention to those APIs.)